com.unity.netcode.gameobjects/Runtime/SceneManagement/SceneEventData.cs

```diff
@@ -245,6 +245,103 @@ internal void InitializeForSynch()
             }
         }
 
+        /// <summary>
+        /// Sorts the synchronization order of the NetworkObjects to be serialized
+        /// by parents before children order
+        /// </summary>
+        private void SortParentedNetworkObjects()
```

Comment:
This seems like a case where doing a breadth-first traversal of the scene graph and building a list by appending each node in the order of traversal would accurately get the ordering you want. I feel like that might be a simpler implementation of this?

Reply:
Yeah... not sure why I didn't use the sort compare approach for the `NetworkObject` children.
I made an adjustment that greatly simplifies the sorting code and should be faster than the first implementation for sure. This only is invoked when handling the initial synchronization _(i.e. called once per synchronizing client)_.